realm/src/main/java/io/realm/internal/NativeObjectReference.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2014 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.internal;
+
+import java.lang.ref.PhantomReference;
+import java.lang.ref.ReferenceQueue;
+
+/**
+ * This phantom reference extension is specialized in holding the pointer to the native objects that needs to be
+ * handled.s
```

Comment:
A bit convoluted. Perhaps something like "This class is used for holding the reference to the native pointers present in NativeObjects. This is required as phantom references cannot access the original objects for this value."

Adding some links to external presentation(s)/doc would also be excellent. These concepts are a bit tricky to get right, even though you make it look easy with so little code ;)